### ISSUE TITLE
Split the Repository

### ISSUE THREAD

**Issue by Samq64** [ScratchAddons/ScratchAddons#6329]
### Why this would be helpful

After #6272 the settings page will need to be built. The plan is to include the builds in the repository so contributors don't need to build anything to work with addons, but I don't think that's ideal since it clutters the Git history.

### My suggestion

Splitting the popup and settings page into it's own repository, and maybe the core and APIs too if #6233 is implemented. This will complicate things for some people but I don't think anything will need to be linked when just working with addons.

### Possible alternatives

Requiring all users to build the settings page and popup to work on addons or including the builds in the repository.

### Additional context

This was discussed briefly on Discord recently.

**Comment by WorldLanguages** [comment]
> including the builds in the repository

This is the current plan on #6272. The built files will be available in the repository. Building will only be needed to modify the settings page.

**Comment by WorldLanguages** [comment]
> The plan is to include the builds in the repository so contributors don't need to build anything to work with addons, but I don't think that's ideal since it clutters the Git history.

Will including the builds in the repository really clutter the Git history? It should only affect PRs to the settings page, I don't think it's a big deal.

**Comment by Samq64** [comment]
> Will including the builds in the repository really clutter the Git history? It should only affect PRs to the settings page, I don't think it's a big deal.

Yeah, it's probably fine and it doesn't really matter if this happens.

**Comment by Hans5958** [comment]
Disagree, unless we move on from this antique development workflow (sans build step et al), I don't think we can separate it.

If we did it, the first thing I would separate it is the addon and the i18n, not what you infer.

**Comment by Samq64** [comment]
> Disagree, unless we move on from this antique development workflow (sans build step et al), I don't think we can separate it.

I forgot that since the other parts of the project would be dependencies contributors would still need to `npm install` anyway.

**Comment by apple502j** [comment]
Oh no. The pain of developing scratch-vm and scratch-gui. Please don't bring this back.

Like, seriously, splitting the code you want to edit a lot and the code you need to test the addon is the worst idea you can have.

**Comment by Samq64** [comment]
> Oh no. The pain of developing scratch-vm and scratch-gui. Please don't bring this back.

Except we won't require super outdated dependencies and possibly different versions of node and npm.

**Comment by DNin01** [comment]
Yeah, I don't know how easy it would be to basically have to merge two Git repositories together. I don't think that would work well with Git.

**Comment by Hans5958** [comment]
> Yeah, I don't know how easy it would be to basically have to merge two Git repositories together. I don't think that would work well with Git.

Introducing [Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules).

**Comment by Samq64** [comment]
> Oh, let me tell you this cool thing called [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules).

How well would that work with this project?

**Comment by Samq64** [comment]
> I have said this in the Discord server already, but I don't think I did here on GitHub: **the settings page should be a separate project with its own maintainers, issues, PRs, and public APIs.** It's just one page, but it's an enormous UI/UX challenge that deserves specific dedication instead of getting covered by other pull requests.
>
> The fact the settings page is so coupled with the rest of the extension has created multiple problems:
> - No innovation and hard to make changes. (Basically, this is what we solve for Scratch thanks to the extension capability in browsers).
> - No customization for the built-in settings page, with the only workaround being downloading from source, because an extension can't affect another extension.
> - Also related to the item above, no competition. For example @iqnite could implement his idea by manually tagging each addon and building a separate settings page. But he won't do so because, surely, it won't get merged, and no one will use it.
> - Other projects that could take advantage of the settings page UI either build their own or have trouble porting it (TurboWarp and SA-Userscript respectively)
>
> This is what I think we should attempt for the long term then:
> - The extension should expose an API to the settings page instead of the settings page calling `chrome.*` methods directly.
> - Localization in the settings page should be handled with our own localization system instead of `chrome.i18n` for more flexibility, including the ability to choose any language for the settings page.
> - At this point it wouldn't matter if the default settings page is Vue, React, Svelte or whatever, it's up to the maintainers of it.
> - This might be the most important point. We can allow alternate setting pages (developed by anyone else) in Scratch Addons as long as they're reviewed for security and privacy. Users can then, if they want, choose which settings page to use. If an alternate settings page is universally agreed to be better in every aspect, it can become the new default and existing users can still rollback to the old one if they want.

_Originally posted by @WorldLanguages in https://github.com/ScratchAddons/ScratchAddons/discussions/7712#discussioncomment-10279895_

**Comment by Samq64** [comment]
There has been ore discussion the dev Discord about this over the last few days, so I'll re-open this.

**Comment by DNin01** [comment]
Like W_L's comment quoted by Samq above, since we're an open-source platform, basically, I think making Scratch Addons more modular would benefit community development. I think it could be broken up into discrete parts — UI, common libraries, and the addons themselves — that, on top of a core layer with the framework and APIs needed for addons to work, come together to make Scratch Addons as a whole, and parts can be swapped out as desired, whether you want non-official addons or a non-official UI to manage your addons.

It's kind of like how apps run on operating systems which use kernels which run on hardware.

**Comment by DNin01** [comment]
Also, we don't have to put each component of Scratch Addons in separate repositories. Many projects use what are called "monorepos" which are single repositories that contain multiple separate packages. Scratch is [migrating to a monorepo structure](https://github.com/scratchfoundation/scratch-editor) right now.
